src/Core/Content/Category/Subscriber/CategorySubscriber.php

```diff
@@ -64,6 +69,12 @@ public function entityLoaded(EntityLoadedEvent $event): void
 
             // mark cms page as set in the subscriber
             $category->setCmsPageIdSwitched(true);
+
+            if ($isSalesChannelEvent && $category instanceof SalesChannelCategoryEntity) {
```

Comment:
This condition should be the first one in the foreach loop. Otherwise, it might be skipped because of the continue statements in the earlier conditions.